lib/vanagon/component.rb

```diff
@@ -290,6 +290,10 @@ def get_build_dir
       end
     end
 
+    def get_dependency_hash
+      { name => { 'version' => version, 'ref' => options[:ref] }.delete_if { |k, v| !v } }
```

Comment:
> Regarding how many hashes are created in the process, do we know the implementation of delete_if? Maybe it creates an internal hash to modify so it doesn't get concurrent modification problems when attempting to iterate and remove elements at the same time.

It's part of Ruby core, so it's written in C. That means it's going to be significantly faster than anything you write in Ruby to do the same thing. But since you asked:
 
From the [doc](https://ruby-doc.org/core-2.2.0/Array.html#method-i-delete_if)

```c
static VALUE
rb_ary_delete_if(VALUE ary)
{
    RETURN_SIZED_ENUMERATOR(ary, 0, 0, ary_enum_length);
    ary_reject_bang(ary);
    return ary;
}
```